report/renderer/report-utils.js

```diff
@@ -187,6 +190,75 @@ class ReportUtils {
     return clone;
   }
 
+  /**
+   * Given an audit's details, identify and return a URL locator function that
+   * can be called later with an `item` to extract the URL of it.
+   * @param {LH.FormattedIcu<LH.Audit.Details.TableColumnHeading[]>} headings
+   * @return {{(item: LH.FormattedIcu<LH.Audit.Details.TableItem>): string|undefined}=}
+   */
+  static getUrlLocatorFn(headings) {
+    // The most common type, valueType=url.
+    const urlKey = headings.find(heading => heading.valueType === 'url')?.key;
+    if (urlKey && typeof urlKey === 'string') {
+      // Return a function that extracts item.url.
+      return (item) => {
+        const url = item[urlKey];
+        if (typeof url === 'string') return url;
```

Comment:
technically table items can override their `valueType` themselves, so this would just opt them out even if they could still be classified from `item[urlKey].value`. Maybe a corner case worth dealing with after #14655, though (not sure if any audit actually does this with `url`)

Reply:
Not that I've seen in any of our audits. Why would we allow that, vs. introducing a new `valueType`?